### ISSUE TITLE
Infinite Matter Issues

### ISSUE THREAD

**Issue by D3athrow-Issues** [d3athrow/vgstation13#17910]
(WEB REPORT BY: minimeatwad REMOTE: 172.93.109.202:7777)
# Revision
4315dc729429be213ad2e4f04eb8f4bde0a7886f

# Description
the ice cream machine usually in the kitchen or bar can create infinite amounts of nutriment and items at no delay, which is fucking insane. the rapid pipe dispenser can also create a infinite amount of items, both can be used to get the smallest size of singularity all the way up to the biggest in a matter of seconds, the ice cream machine possibly having the potential to lag the server out and definitely needs fixing.

**Comment by ghost** [comment]
Yeah, ice cream's not the best but when you can trivially make 4,000 25 nutriment watermelons in a single life cycle of a soil farm you stop caring because it's just a time gate that's easily met while they otherwise setup their gimmick. I guess if you wanted to split hairs, an incremental improvement could be having a massive 2,500 or whatever nutriment reservoir and an internal grinder that auto-ejects all other chemicals (from the grind) into condibottles or whatever.

RPD is something that got abstracted for quality of life and there isn't a neat solution unless you massively overhaul the entire mineral metagame to everyone's loss or it's basically still abstracted and you're just pissing people off when they go build stuff cause limitations wont stop a guy with 6 RPD's at his disposal. Heck, there's a lot of stuff you can do to really speed it up even if it's solely just setting the PA to 3 Yes, it can make loose goose but so can a crate of BoH's get it to fug off and people can just fug off to the outposts if they don't want to leave like usual.

**Comment by Probe1** [comment]
.. you could just make RPDs have a battery and use up charge on every item.

**Comment by SonixApache** [comment]
Please, for the love of fuck, do not arbitrarily limit the RPD.

The ice cream machine needs a refactor, this has been discussed and agreed on time and time again.
See also: #16820

**Comment by abagone** [comment]
If you're really worried about somebody expanding the singularity with the RPD just make pipes give **NO MASS** to the singularity.

If the RPD did use power it should use **1 power for any item** and start with a **10,000 power cell** which you can replace and upgrade if you wish and it should **recharge by moving around** and look it technically uses power but it basically doesn't matter is everyone happy now? 
But honestly coding this would basically just be a waste of time. 

Speaking of infinite matter I'll have to test this but doesn't the material synthesizer generate more than enough metal and glass to make more matter cartridges for itself? 
I know that the mommi one uses power and can scan just about anything including **diamonds**!

Not that you should change that I'm pretty sure the cyborg synthesizer was **nerfed** to prevent people from ordering the cyborgs to print diamonds and sit in cyborg charger but mommi will most just make material they need and keep it in their nest **it's not like they can be ordered to give you diamonds** unlike the cyborg.

**Comment by MiniMeatwad** [comment]
the ice cream machine has no cooldown for making ice cream which means it can be spammed with a macro to lag the ever living fuck out of people if not the server

**Comment by MiniMeatwad** [comment]
the ornaments in the autolathe create glass shards but cost nothing to make since the box can be recycled for all the glass back from making it in the first place, with the glass shards creating more glass than the box costed in the first place too, the jar also still creates shards whilst costing less than shards

**Comment by MadmanMartian** [comment]
The fix for this was closed
